tuf/ngclient/updater.py

```diff
@@ -205,17 +205,21 @@ def download_target(
         else:
             target_base_url = _ensure_trailing_slash(target_base_url)
 
-        full_url = parse.urljoin(target_base_url, targetinfo["filepath"])
+        target_filepath = targetinfo["filepath"]
+        target_fileinfo = targetinfo["fileinfo"]
```

Comment:
```suggestion
        target_fileinfo:TargetFile = targetinfo["fileinfo"]
```
the real fix would be fixing the targetinfo argument type but this helps typing in the mean time